scripts/auth.js

```diff
@@ -0,0 +1,184 @@
+  }
+  function validateNickname() {
+    const nicknameValue = nicknameInput.value.trim();
+    let errorMessage = nicknameInput.closest(".input-group").querySelector(".error-message");
+    if (!nicknameValue) {
+      nicknameInput.classList.add("input-error");
+      if (!errorMessage) {
+        errorMessage = document.createElement("span");
+        errorMessage.className = "error-message";
+        errorMessage.style.cssText = "color: red; font-size: 14px; display: block; margin-top: 4px;";
+        errorMessage.innerText = "닉네임을 입력해주세요.";
+        nicknameInput.closest(".input-group").appendChild(errorMessage);
+      } else {
+        errorMessage.innerText = "닉네임을 입력해주세요.";
+      }
+    } else {
+      nicknameInput.classList.remove("input-error");
+      if (errorMessage) errorMessage.remove();
+    }
+    checkFormValidity();
+  }
+
+  // 비밀번호 - 회원가입
+  const passwordCheckInput = document.getElementById("passwordCheck");
+  if (passwordCheckInput) {
+    passwordCheckInput.addEventListener("blur", validatePasswordCheck);
+    passwordCheckInput.addEventListener("input", validatePasswordCheck);
+  }
+  function validatePasswordCheck() {
+    if (passwordInput && passwordCheckInput) {
+      const passwordValue = passwordInput.value;
+      const passwordCheckValue = passwordCheckInput.value;
+      let errorMessage = passwordCheckInput.closest(".input-group").querySelector(".error-message");
+      if (passwordValue !== passwordCheckValue) {
+        passwordCheckInput.classList.add("input-error");
+        if (!errorMessage) {
+          errorMessage = document.createElement("span");
+          errorMessage.className = "error-message";
+          errorMessage.style.cssText = "color: red; font-size: 14px; display: block; margin-top: 4px;";
+          errorMessage.innerText = "비밀번호가 일치하지 않습니다.";
+          passwordCheckInput.closest(".input-group").appendChild(errorMessage);
+        } else {
+          errorMessage.innerText = "비밀번호가 일치하지 않습니다.";
+        }
+      } else {
+        passwordCheckInput.classList.remove("input-error");
+        if (errorMessage) errorMessage.remove();
+      }
+      checkFormValidity();
+    }
+  }
+
+  // 비밀번호 보이게 하기
+  const passwordWrappers = document.querySelectorAll(".password-wrapper");
+  passwordWrappers.forEach(function (wrapper) {
+    const toggleButton = wrapper.querySelector("button");
+    const input = wrapper.querySelector("input");
+    if (toggleButton && input) {
+      toggleButton.addEventListener("click", function (e) {
+        e.preventDefault();
+        input.type = input.type === "password" ? "text" : "password";
+      });
+    }
+  });
+
+  // 폼 제출 처리
+  form.addEventListener("submit", function (e) {
+    e.preventDefault();
+    if (button.disabled) return;
+    if (nicknameInput || passwordCheckInput) {
+      window.location.href = "login.html";
+    } else {
+      window.location.href = "/items";
+    }
+  });
```

Comment:
💊 제안
form이 유효하지 않으면 submit button이 disabled 상태라 제출이 되지 않으므로 관련 코드는 없어도 될 것 같아요~

```suggestion
  form.addEventListener("submit", function (e) {
    e.preventDefault();
    
    if (nicknameInput || passwordCheckInput) {
      window.location.href = "login.html";
    } else {
      window.location.href = "/items";
    }
  });
```